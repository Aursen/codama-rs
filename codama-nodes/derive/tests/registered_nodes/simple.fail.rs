use codama_nodes_derive::RegisteredNodes;

pub struct NumberTypeNode {}
pub struct StructTypeNode {}
pub struct StructFieldTypeNode {}

#[derive(RegisteredNodes)]
pub enum RegisteredTypeNode {
    Number(NumberTypeNode),
    Struct(StructTypeNode),
    #[registered]
    StructField(StructFieldTypeNode),
}

fn main() {
    TypeNode::StructField(StructFieldTypeNode {});
}
