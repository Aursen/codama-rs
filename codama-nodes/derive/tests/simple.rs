use codama_nodes_derive::IntoEnum;

pub struct NumberTypeNode {}
pub struct StringTypeNode {}

#[derive(IntoEnum)]
pub enum TypeNode {
    Number(NumberTypeNode),
    String(StringTypeNode),
}

fn main() {
    let number: TypeNode = NumberTypeNode {}.into();
    assert!(matches!(number, TypeNode::Number(_)));

    let string: TypeNode = StringTypeNode {}.into();
    assert!(matches!(string, TypeNode::String(_)));
}
