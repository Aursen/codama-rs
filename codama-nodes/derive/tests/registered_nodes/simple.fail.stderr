error[E0599]: no variant or associated item named `StructField` found for enum `TypeNode` in the current scope
  --> tests/registered_nodes/simple.fail.rs:16:15
   |
8  | pub enum RegisteredTypeNode {
   | --------------------------- variant or associated item `StructField` not found for this enum
...
16 |     TypeNode::StructField(StructFieldTypeNode {});
   |               ^^^^^^^^^^^ variant or associated item not found in `TypeNode`
